core/src/schema/namespace.rs

```diff
@@ -160,6 +166,78 @@ impl Namespace {
             Err(failed!(target: Release, NotFound => "no such collection: '{}'{}", name, suggest))
         }
     }
+    fn get_ajd_list(&self) -> Vec<(Name, Name)> {
+        let mut list = Vec::new();
+        for (n, c) in self.clone() {
+            match c {
+                Content::SameAs(same_as_content) => {
+                    list.push((same_as_content.ref_.collection().clone(), n));
+                }
+                _ => {}
+            }
+        }
+        list
+    }
+
+    pub fn topo_sort(&self) -> Option<Vec<Name>> {
+        eprintln!("namespace: {:?}", self);
+        let lists: Vec<(Name, Name)> = self.get_ajd_list();
+        let mut q: VecDeque<Name> = VecDeque::new();
+        //topologically sorted vec of names to be returned
+        let mut sorted: Vec<Name> = Vec::new();
+
+        //or graph
+        let mut graph: NameGraph = HashMap::new();
```

Comment:
I think you can use ```NameGraph::new();``` here to initialize the variable instead of ```HashMap::new();```, it seems cleaner.